### PR TITLE
[DRAFT] Autogenerate configuration file based on data.inventory

### PR DESCRIPTION
**Pull request by jpreese**
@garethahealy 

An initial proof of concept (yes, it's ugly!) for #16 -- the `config.yaml` was automatically generated based on the src rego.

_Need:_ 
- Flag to set the output directory of the config file. 
- Cluster-scoped resources
- APIGroups? Have really seen any in the examples.

_Current uncertainty:_

While this seems to make sense on paper, I'm wondering if manually maintaining the config is that big of an inconvenience to introduce something like this. My primary concern would be policies in _another_ repository that might use `data.inventory`, but there is only ever a single config. Is this even an issue?

### CODE REVIEW

**Comment by garethahealy** [comment]
from a quick test, the output looks good for me.

File: internal/commands/create.go [hunk 1]
```diff
@@ -134,9 +142,91 @@ func runCreateCommand(path string) error {
...
 
+	syncResources, err := getSyncResources(policies)
+	if err != nil {
+		return fmt.Errorf("get sync resources: %w", err)
+	}
+
+	config := getConfig(syncResources)
+	configBytes, err := yaml.Marshal(&config)
+	if err != nil {
+		return fmt.Errorf("marshal constraint: %w", err)
```
Comment (jalseth) [inline internal/commands/create.go:153]: constraint -> config

**Comment by jpreese** [comment]
@jalseth and I talked about this briefly offline, and I think we're of the opinion that it feels a bit early for this. While it'd be nice to generate the config off of the policies, and possible in a perfect scenario, it should not _that_ difficult to manage the configuration outside of Konstraint.

I'd rather wait until the Gatekeeper team flushes out this behavior more rather than trying to keep up with their changes.

**Comment by jalseth** [comment]
We are in agreement, so closing this for now. We will plan on revisiting this once the Gatekeeper config CRD moves into beta.

**Pull request closed without merging**